source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -0,0 +1,117 @@
+#include "MaterialXData.h"
+#include "Plugin.h"
+#include "Util.h"
+
+#include <MaterialXFormat/XmlIo.h>
+#include <MaterialXGenGlsl/GlslShaderGenerator.h>
+#include <MaterialXGenShader/Util.h>
+
+#include <maya/MViewport2Renderer.h>
+#include <maya/MFragmentManager.h>
+
+MaterialXData::MaterialXData(const std::string& materialXDocument, const std::string& elementPath)
+{
+	libSearchPath = Plugin::instance().getLibrarySearchPath();
+	createDocument(materialXDocument);
+
+	if (doc)
+	{
+		element = doc->getDescendant(elementPath);
+	}
+
+	MaterialX::GenContext* glslContext = new MaterialX::GenContext(MaterialX::GlslShaderGenerator::create());
+
+	// Stop emission of environment map lookups.
+	glslContext->getOptions().hwSpecularEnvironmentMethod = MaterialX::SPECULAR_ENVIRONMENT_NONE;
+	glslContext->registerSourceCodeSearchPath(libSearchPath);
+	contexts.push_back(glslContext);
+
+	glslFragmentWrapper = new MaterialX::OGSXMLFragmentWrapper(glslContext);
+	glslFragmentWrapper->setOutputVertexShader(true);
+
+}
+
+MaterialXData::~MaterialXData()
+{
+	if (glslFragmentWrapper)
+	{
+		delete glslFragmentWrapper;
+	}
+}
+
+void MaterialXData::createDocument(const std::string& materialXDocument)
+{
+	// Create document
+	doc = MaterialX::createDocument();
+	MaterialX::readFromXmlFile(doc, materialXDocument);
+
+	// Load libraries
+	const MaterialX::StringVec libraries = { "stdlib", "pbrlib", "bxdf", "stdlib/genglsl", "pbrlib/genglsl" };
+	MaterialX::loadLibraries(libraries, libSearchPath, doc);
+}
+
+bool MaterialXData::isValidOutput()
+{
+	if (!element) return false;
```

Comment:
Nitpicky item for Pavlo's sake. We should use syntax 
```
if (condition)
{
}
```
formatting.

Reply:
OK. BTW could you point me at the coding conventions please?